### PR TITLE
Support PATCH method

### PR DESCRIPTION
**Pull request by nronas**

### CODE REVIEW

**Review by angetheo — APPROVED** [review]
perfect!

File: spec/routemaster/api_client_spec.rb [hunk 1]
```diff
@@ -48,6 +56,15 @@
       end
     end
 
+    context 'PUT request' do
```
Comment (mezis) [inline spec/routemaster/api_client_spec.rb]: this says PUT, but you're testing PATCH ?
(couple lines below too)

File: lib/routemaster/drain.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 module Routemaster
   module Drain
-    VERSION = '2.0.0'
+    VERSION = '2.1.0'
```
Comment (mezis) [inline lib/routemaster/drain.rb]: Usually it's considered bad form to bump version in a branch, but as long as there are no other outstanding PRs I won't insist :)

File: lib/routemaster/api_client.rb [hunk 3]
```diff
@@ -33,7 +33,18 @@ def post(url, body: {}, headers: {})
...
+        end
+      end
+    end
+
+    def patch(url, body: {}, headers: {})
+      host = URI.parse(url).host
+      response_wrapper do
+        connection.patch do |req|
+          req.url url
+          req.headers = headers.merge(auth_header(host))
```
Comment (mezis) [inline lib/routemaster/api_client.rb:46]: This gets repeated everywhere; we'll probably want to extract it to a middleware eventually.

**Comment by mezis** [comment]
Minor gripe in the tests, otherwise LGTM.

**Comment by FriedSock** [comment]
LGTM 👍

File: spec/routemaster/api_client_spec.rb (hunk 1 above)
Comment (nronas) [reply spec/routemaster/api_client_spec.rb]: 👍

File: lib/routemaster/drain.rb (hunk 2 above)
Comment (nronas) [reply lib/routemaster/drain.rb]: yeap that's correct. I should merge and bump on master instead on branch

File: lib/routemaster/api_client.rb (hunk 3 above)
Comment (nronas) [reply lib/routemaster/api_client.rb:46]: 👍  Noted!

**Pull request merged**